homeassistant/components/androidtv/media_player.py

```diff
@@ -168,180 +168,110 @@
 }
 
 
-def setup_androidtv(hass, config):
-    """Generate an ADB key (if needed) and load it."""
-    adbkey = config.get(CONF_ADBKEY, hass.config.path(STORAGE_DIR, "androidtv_adbkey"))
-    if CONF_ADB_SERVER_IP not in config:
-        # Use "adb_shell" (Python ADB implementation)
-        if not os.path.isfile(adbkey):
-            # Generate ADB key files
-            keygen(adbkey)
-
-        # Load the ADB key
-        signer = ADBPythonSync.load_adbkey(adbkey)
-        adb_log = f"using Python ADB implementation with adbkey='{adbkey}'"
-
-    else:
-        # Use "pure-python-adb" (communicate with ADB server)
-        signer = None
-        adb_log = f"using ADB server at {config[CONF_ADB_SERVER_IP]}:{config[CONF_ADB_SERVER_PORT]}"
-
-    return adbkey, signer, adb_log
-
-
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info=None,
+) -> None:
     """Set up the Android TV / Fire TV platform."""
-    hass.data.setdefault(ANDROIDTV_DOMAIN, {})
 
-    address = f"{config[CONF_HOST]}:{config[CONF_PORT]}"
+    host = config[CONF_HOST]
 
-    if address in hass.data[ANDROIDTV_DOMAIN]:
-        _LOGGER.warning("Platform already setup on %s, skipping", address)
-        return
-
-    adbkey, signer, adb_log = await hass.async_add_executor_job(
-        setup_androidtv, hass, config
-    )
+    # get main data
+    config_data = {
+        CONF_HOST: host,
+        CONF_DEVICE_CLASS: config.get(CONF_DEVICE_CLASS, DEFAULT_DEVICE_CLASS),
+        CONF_PORT: config.get(CONF_PORT, DEFAULT_PORT),
+    }
+    for key in (CONF_ADBKEY, CONF_ADB_SERVER_IP, CONF_ADB_SERVER_PORT, CONF_NAME):
+        if key in config:
+            config_data[key] = config[key]
+
+    # get options
+    config_options = {}
+    for key in (
+        CONF_APPS,
+        CONF_EXCLUDE_UNNAMED_APPS,
+        CONF_GET_SOURCES,
+        CONF_SCREENCAP,
+        CONF_STATE_DETECTION_RULES,
+        CONF_TURN_OFF_COMMAND,
+        CONF_TURN_ON_COMMAND,
+    ):
+        if key in config:
+            if key == CONF_STATE_DETECTION_RULES:
+                config_options[key] = json.dumps(config[key])
```

Comment:
Why is this stored as a json string?

Reply:
This is a quite complex configuration (see [here](https://www.home-assistant.io/integrations/androidtv/#custom-state-detection)). The only way that I found to manage this as config option is to save it as json string and validating the content with a function provided by the library.

Reply:
One alternative would be to let them provide a state file, but at that point its seems better to leave them in yaml.

Let's ask for another set of eyes on this.